src/Features/Core/Shared/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -2,102 +2,64 @@
 
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.Diagnostics.EngineV1;
 using Roslyn.Utilities;
-using System;
 
 namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static class DiagnosticAnalyzerExtensions
     {
-        public static async Task<DiagnosticAnalyzerCategory> GetDiagnosticAnalyzerCategoryAsync(this DiagnosticAnalyzer analyzer, DiagnosticAnalyzerDriver driver)
+        public static DiagnosticAnalyzerCategory GetDiagnosticAnalyzerCategory(this DiagnosticAnalyzer analyzer)
```

Comment:
@heejaechang You may want to take a look at this. The heuristic to determine if an analyzer supports incremental method body analysis has now been replaced with the following logic:
1. IBuiltInAnalyzer has a new method `GetAnalyzerCategory`, so our built in IDE analyzers explicitly state whether they report syntax/semantic/compilation diagnostics and whether or not they can be run incrementally for method body edits.
2. For rest of the public analyzers, we conservatively assume they can generate any kind of diagnostic and do not support incremental method body analysis.